tests/lib/devicetree/api/src/main.c

```diff
@@ -1639,11 +1639,20 @@ static void test_clocks(void)
 		      25000000, "");
 }
 
+#undef DT_DRV_COMPAT
+#define DT_DRV_COMPAT vnd_spi_device
 static void test_parent(void)
 {
 	/* The label of a child node's parent is the label of the parent. */
 	zassert_true(!strcmp(DT_LABEL(DT_PARENT(TEST_SPI_DEV_0)),
 			     DT_LABEL(TEST_SPI_BUS_0)), "");
+
+	/*
+	 * Same previous check now using first instance of vnd,spi-device,
+	 * child of TEST_SPI_BUS_0.
+	 */
+	zassert_true(!strcmp(DT_LABEL(DT_INST_PARENT(0)),
+			     DT_LABEL(TEST_SPI_BUS_0)), "");
```

Comment:
Can you use `TEST_SPI` instead of `TEST_SPI_BUS_0` here? They are the same node identifier, but the way it is currently written makes it look like instance 0 is `TEST_SPI_DEV_0`, which is not necessarily true. Using `TEST_SPI` instead is a little cleaner, I think. That makes it clear that it doesn't matter which instance of `vnd,spi-device` is instance 0, since they both happen to have the same parent node.

Otherwise LGTM.